### PR TITLE
CXP-1074: Apps + OpenID Connect

### PR DESCRIPTION
**Pull request by tseho**
Screen Shot 2022-01-25 at 15 18 02-fullpage

### CODE REVIEW

File: content/apps/using-openid.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+| `aud`       | Id of the OAuth 2.0 client           |
+| `iat`       | Timestamp of token creation          |
+| `exp`       | Timestamp of token expiration        |
+| `email`     | **(optional) Email of the user**     |
+| `firstname` | **(optional) Firstname of the user** |
+| `lastname`  | **(optional) Lastname of the user**  |
+
+Additonal information (`email`, `firstname`, `lastname`, ...) are only present if you requested the corresponding
+authentication scopes and those were approved by the user.
+
```
Comment (AymericPlanche) [inline content/apps/using-openid.md:107]: I would add a notice like: "email, firstname and lastname are values that can be edited on the PIM side, they can change over time. The **subject**  claim (aka `sub`) is a unique and immutable technical identifier which is shared among the PIM and the App. You can map this technical identifier coming from the PIM to your user management stack.


:arrow_up: do adapt, but you see what I mean

File: content/apps/using-openid.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+The ID Token sent by Akeneo PIM contains a signature, and you must verify it to guarantee that the payload has not be 
+tampered with.
+
+To validate the signature, you must retrieve the public key available at the url `{PIM}/connect/apps/v1/openid/public-key`.  
+
+Then, follow the instructions of the library you are using.
+
+::: warning
+The pair of private/public keys is regenerated regularly. You should always retrieve the lastest public key when validating a signature.
```
Comment (AymericPlanche) [inline content/apps/using-openid.md:118]:
```suggestion
The pair of private/public keys is regenerated regularly for security reasons. You should always retrieve the lastest public key when validating a signature.
```

File: content/apps/using-openid.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+If an user is trying to access your App from his Akeneo PIM, and you want to authenticate him, start an
+[Authorization Request](/apps/using-oauth2.html#authorization-request),
+**even if you already are connected to his Akeneo PIM**.
+
+**During this new Authorization Request, you must request all the scopes your App needs**, including the Authorization scopes,
+in addition of the OpenID scopes.  
+
+## Extracting user information from the ID Token
+
+At the end of the OAuth 2.0 process, if your Access Token Request is accepted, you will receive a JSON response with both tokens:
```
Comment (AymericPlanche) [inline content/apps/using-openid.md:53]: note that the `access_token` you'll get with the authentication request, is the same `access_token` you get during the **authorization** workflow.

**Review by AymericPlanche — APPROVED** [review]
I know it's hard to find the right balance between saying too much and add complexity in comprehension AND do not document "obvious" details. 

pick what you want from my returns ;)

**Review by pchasle — APPROVED** [review]

File: content/apps/using-openid.md (hunk 1 above)
Comment (tseho) [reply content/apps/using-openid.md:107]: I agree with the first part, I think the first part is overwhelming for consumers of this documentation.
I preferred to be concise and said `sub` is the unique user id.

File: content/apps/using-openid.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+OpenID Connect is a simple identity layer on top of the OAuth 2.0 protocol.  
+See [OpenID official website](https://openid.net/connect/) for more info.
+:::
+
+Basically, you use the same process as for Authorization, but you request an additional scope
+and you will receive, alongside the Access Token, an ID Token containing the information of the current user.
+
+![App authentication diagram](../img/apps/app-authentication-sequence-diagram.png)
+
+## Authenticate during the first Authorization request
```
Comment (RiffVentura) [inline content/apps/using-openid.md:24]:
```suggestion
## Authentication during the first Authorization request
```
and similar titles

Comment (tseho) [reply content/apps/using-openid.md:24]: I will check the english with Camille, I'm expecting a review on the content from you ;)

**Review by RiffVentura — APPROVED** [review]

File: content/apps/using-openid.md (hunk 1 above)
Comment (tseho) [reply content/apps/using-openid.md:107]: I went with

```
`email`, `firstname` and `lastname` are values that can be edited on Akeneo PIM and are not verified by
Akeneo. The only value that will truly identify an user is his unique user id in the subject claim (`sub`). 
```

File: content/apps/using-openid.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  "access_token": "Y2YyYjM1ZjMyMmZlZmE5Yzg0OTNiYjRjZTJjNjk0ZTUxYTE0NWI5Zm",
+  "token_type": "bearer",
+  "scope": "openid email profile read_products write_products",
+  "id_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJodHRwOi8vM.XcmmANmSC2RHqWOI"
+}
+```
+
+### Decoding the ID Token
+
+An ID Token is a JWT is composed of 3 parts encoded independently in base64: `header.payload.signature`.
```
Comment (masacc) [inline content/apps/using-openid.md:68]:
```suggestion
An ID Token is a JWT that is composed of 3 parts encoded independently in base64: `header.payload.signature`.
```

**Review by masacc — APPROVED** [review]

**Pull request merged**